### PR TITLE
Fix atomic operation compatibility for ARMv8-A (Raspberry Pi 4) by adjusting compilation flags

### LINKED ISSUE #146792: `Illegal instruction (core dumped)` on Raspberry Pi 4 when exporting ONNX with `torch 2.6.0`
#### **Description**
On Raspberry Pi 4, `torch.onnx.export` fails with `Illegal instruction (core dumped)` in `torch 2.6.0`. The same code works fine on `torch 2.5.1`. The issue occurs when using `x.expand(x.shape[0], -1, -1)` inside a `torch.nn.Module`. The crash happens **only during ONNX export**, not during regular inference.

#### **Code to Reproduce**
```python
import torch

class Module(torch.nn.Module):
    def forward(self, x):
        return x.expand(x.shape[0], -1, -1)  # Crashes here during ONNX export

model = Module()
dummy_inputs = tuple(torch.randn(1, 1, 192))

# Running the model works fine
res = model(*dummy_inputs) 

# Exporting to ONNX causes core dump
torch.onnx.export(model, opset_version=20, f="./m.onnx", args=dummy_inputs) 
```

#### **Error Output**
```
Illegal instruction (core dumped)
```

#### **Device and Environment Details**
| Device                     | PyTorch Version | Execution Type | Status  |
|----------------------------|----------------|----------------|---------|
| MacBook Pro M4 (native)    | 2.6.0          | Native         | ✅ Works |
| MacBook Pro M4 (Docker)    | 2.6.0          | Docker         | ✅ Works |
| Raspberry Pi 4 (native)    | 2.5.1          | Native         | ✅ Works |
| Raspberry Pi 4 (Docker)    | 2.5.1          | Docker         | ✅ Works |
| Raspberry Pi 4 (native)    | 2.6.0          | Native         | ❌ **Fails** |
| Raspberry Pi 4 (Docker)    | 2.6.0          | Docker         | ❌ **Fails** |
| Raspberry Pi 5 (native)    | 2.6.0          | Native        | ✅ Works |

# raspi 4 vs 5 cpu Features
running `cat /proc/cpuinfo | grep 'Fe' | uniq`
## raspi 4
```bash
Features	: fp asimd evtstrm crc32 cpuid
```
## raspi  5
```bash 
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm lrcpc dcpop asimddp
```

### Versions

Collecting environment information...
PyTorch version: 2.6.0+cpu
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Debian GNU/Linux 12 (bookworm) (aarch64)
GCC version: Could not collect
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.36

Python version: 3.11.11 (main, Feb  4 2025, 13:44:55) [GCC 12.2.0] (64-bit runtime)
Python platform: Linux-5.15.32-v8+-aarch64-with-glibc2.36
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                    aarch64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
CPU(s):                          4
On-line CPU(s) list:             0-3
Vendor ID:                       ARM
Model name:                      Cortex-A72
Model:                           3
Thread(s) per core:              1
Core(s) per cluster:             4
Socket(s):                       -
Cluster(s):                      1
Stepping:                        r0p3
CPU(s) scaling MHz:              100%
CPU max MHz:                     1800.0000
CPU min MHz:                     600.0000
BogoMIPS:                        108.00
Flags:                           fp asimd evtstrm crc32 cpuid
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Spec store bypass: Vulnerable
Vulnerability Spectre v1:        Mitigation; __user pointer sanitization
Vulnerability Spectre v2:        Vulnerable
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected

Versions of relevant libraries:
[pip3] numpy==1.26.4
[pip3] onnx==1.16.1
[pip3] onnxruntime==1.19.2
[pip3] onnxruntime_extensions==0.13.0
[pip3] torch==2.6.0
[pip3] torchvision==0.21.0
[pip3] uni_pytorch==0.0.0
[conda] Could not collect

cc @ezyang @gchanan @zou3519 @kadeng @msaroufim @malfet @snadampal @milpuz01

### PR DESCRIPTION
**Pull request by maajidkhann**
**Issue:**
* The ldaddal instruction is an AArch64 atomic operation available from ARMv8.1-A onwards.
* Raspberry Pi 4 (Cortex-A72) is ARMv8-A, which does not support ldaddal, leading to failures when running PyTorch built with march=armv8.2-a+sve
* This led to an issue when running PyTorch on ARMv8-A (Raspberry Pi 4), as unsupported atomic operations were generated.

**Fix:**
* Updated the build flags to explicitly use **-march=armv8-a+sve**, ensuring GCC and clang promotes it correctly and resolves compatibility issues with armv8 and still work correctly for SVE like before.
* This ensures that PyTorch builds correctly for ARMv8-A platforms (e.g., Raspberry Pi 4) while still enabling SVE for supported hardware.

Test plan:
 - Allocate `a1.4xlarge` on AWS
 - Run following script using wheel produced by this PR
 ```python
import torch
def f(x):
    return x.sin() + x.cos()

print(torch.__version__)
f_c = torch.jit.script(f)
```
- Observe no crash
```
$ python3 foo.py 
2.7.0.dev20250313+cpu
```
- Observe crash with 2.6.0
```
$ python3 foo.py 
2.6.0+cpu
Illegal instruction (core dumped)
```

Fixes #146792

cc @malfet @snadampal @milpuz01

### CODE REVIEW

**Comment by maajidkhann** [comment]
Please add the label "ciflow/linux-aarch64

**Comment by maajidkhann** [comment]
@malfet Can you please review this.

**Review by malfet — APPROVED** [review]
LGTM (need to check if it works though), but it fixes the symptom, but not the problem (code sharing between CPU architectures)
